Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsUISlider.cs

```diff
@@ -0,0 +1,835 @@
+                {
+                    _connectedButton = physicalHandsButton;
+                }
+            }
+        }
+
+        #region Unity Methods
+
+        /// <summary>
+        /// Initializes the slider when the GameObject is enabled.
+        /// </summary>
+        private void Start()
+        {
+            Initialize();
+        }
+
+        private void OnDisable()
+        {
+            if(_connectedButton != null)
+            {
+                _connectedButton.OnButtonPressed.RemoveListener(ButtonPressed);
+                _connectedButton.OnButtonUnPressed.RemoveListener(ButtonUnPressed);
+            }
+        }
+
+        private void OnEnable()
+        {
+            if (_connectedButton != null)
+            {
+                _connectedButton.OnButtonPressed.AddListener(ButtonPressed);
+                _connectedButton.OnButtonUnPressed.AddListener(ButtonUnPressed);
+            }
+        }
+
+
+
+        /// <summary>
+        /// Update method called once per frame.
+        /// </summary>
+        private void Update()
+        {
+            // Calculate the change in position of the slider object from its zero position
+            _axisChangeFromZero.x = _slideableObject.transform.localPosition.x - _sliderXZeroPos;
+            _axisChangeFromZero.z = _slideableObject.transform.localPosition.z - _sliderZZeroPos;
+
+            // Calculate the slider value based on the change in position
+            _sliderValue = CalculateSliderValue(_axisChangeFromZero);
+
+            // Check if the slider value has changed
+            if (_prevSliderValue != _sliderValue)
+            {
+                // Check if the slider was released last frame
+                if (_sliderReleasedLastFrame)
+                {
+                    // Snap to segment if applicable
+                    switch (_sliderType)
+                    {
+                        case SliderType.ONE_DIMENSIONAL:
+                            {
+                                if (_numberOfSegments != 0)
+                                {
+                                    SnapToSegment();
+                                }
+                                break;
+                            }
+                        case SliderType.TWO_DIMENSIONAL:
+                            {
+                                if (_twoDimNumberOfSegments.x != 0 || _twoDimNumberOfSegments.y != 0)
+                                {
+                                    SnapToSegment();
+                                }
+                                break;
+                            }
+                    }
+
+                    // Reset flag and update previous slider value
+                    _sliderReleasedLastFrame = false;
+                    _prevSliderValue = _sliderValue;
+                }
+                // Send slider change event
+                SendSliderEvent(SliderChangeEvent, TwoDimSliderButtonPressedEvent);
```

Comment:
what about 1d sliders?

Reply:
was actually the 2d sliders broken here, meant to be sending a changed event, send slider event sends either event depending on the type of slider